packages/components/src/DateTimePickers/pickers/DatePicker/DatePicker.test.js

```diff
@@ -1,14 +1,572 @@
 import React from 'react';
 import { shallow } from 'enzyme';
-
+import isSameDay from 'date-fns/is_same_day';
+import isToday from 'date-fns/is_today';
 import DatePicker from './DatePicker.component';
 
+jest.mock('date-fns/is_today');
+
 describe('DatePicker', () => {
+	function mockIsTodayWith(newToday) {
+		isToday.mockImplementation(date =>
+			isSameDay(date, newToday));
+	}
+
+	beforeEach(() => {
+		mockIsTodayWith(new Date(2018, 5, 20));
+	});
+
 	it('should render a DatePicker', () => {
-		// when
-		const wrapper = shallow(<DatePicker />);
+		const calendar = {
+			year: 2018,
+			monthIndex: 5,
+		};
+
+		const disabledDates = [
+			new Date(2018, 5, 6),
+			new Date(2018, 5, 15),
+		];
+		const selectedDate = new Date(2018, 5, 12);
+
+		const wrapper = shallow(<DatePicker
+			calendar={calendar}
+			selectedDate={selectedDate}
+			onSelect={() => {}}
+			disabledRules={disabledDates}
+		/>);
 
-		// then
 		expect(wrapper.getElement()).toMatchSnapshot();
 	});
+
+	describe('body calendar', () => {
+		describe('first date in the correct day column (Monday as first column)', () => {
+			function getFirstRowCellText(wrapper, column) {
+				return wrapper
+					.find('.theme-calendar-body-row')
+					.first()
+					.find('.theme-calendar-item')
+					.at(column - 1)
+					.find('DayPickerAction')
+					.prop('label');
+			}
+
+			it('should have monday in the 1st column for january 2018', () => {
+				const calendar = {
+					year: 2018,
+					monthIndex: 0,
+				};
+
+				const wrapper = shallow(<DatePicker
+					calendar={calendar}
+					onSelect={() => {}}
+				/>);
+
+				expect(getFirstRowCellText(wrapper, 1)).toBe('1');
+			});
+
+			it('should have thursday in the 4th column for march 2018 ', () => {
```

Comment:
Hmm not sure what you test in the 3 tests here
DayOfWeek position does not depends on the month/year.

Reply:
I renamed some tests.
What I test here is the column position of the first day of month accordingly to its day of week.
=> If first day is a friday, it must be in 5th column

Reply:
✔